GVFS/GVFS.Common/Git/DiffTreeResult.cs

```diff
@@ -167,6 +171,23 @@ public static DiffTreeResult ParseFromLsTreeLine(string line, string repoRoot)
             }
         }
 
+        /// <summary>
+        /// Checks the type (e.g. blob, tree) of an entry from ls-tree output matches
+        /// the type string passed in.
+        /// </summary>
+        /// <param name="line">A line of output from ls-tree</param>
+        /// <param name="typeMarker">The type being checked for (e.g. "blob", "tree")</param>
+        /// <returns>True if type of entry matches type specified.</returns>
+        public static bool IsLsTreeLineOfType(string line, string typeMarker)
+        {
+            if (line.Length <= TypeMarkerStartIndex + typeMarker.Length)
+            {
+                return false;
+            }
+
+            return line.IndexOf(typeMarker, TypeMarkerStartIndex, typeMarker.Length, StringComparison.OrdinalIgnoreCase) >= 0;

```

Comment:
This is the crux of the fix (only look for "tree" at the expected location, not in the name field), but it's not very clear that that's what's happening here. I would change the condition to `== TypeMarkerStartIndex`, not `>=0`, to make that more explicit.